### DISCUSSION TITLE
How to force CSR with app router ?

### DISCUSSION THREAD

**Discussion by LucasStbnr** [vercel/next.js#58973] [Help]
### Summary

Hello,
I'm using NextJS 14 with app router and I would like to force CSR on one page but I can't find out how to disable the prerender on the server.
I need to use localStorage on my page but I cannot use it in a useEffect and when I use it outside my useEffect I have an error.

**Comment by icyJoseph** [comment]
🤔 Why not using `useEffect`? You'd have to do something like this, to deactivate ssr:

```tsx
// page.tsx
'use client'
import {useEffect, useState} from "react";

export default function Page() {
  const [isServer,setServer] = useState(true);
  useEffect(setServer,[]);

  if (isServer) return null;

  return (
     <ComponentThatNeedsTheBrowser />
  );
}
```

- https://demystifying-rsc.vercel.app/client-components/no-ssr/
- https://demystifying-rsc.vercel.app/client-components/next-dynamic/

I am curious though, why can't use do it properly, using `use-effect`?

**Comment by alirabieey1504** [comment]
you can use dynamic inport in next.js  
https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading
